File: python/common/DataAnalyzer.py
import csv

PATH = "../../resources/"


def load(file):
    with open(PATH + file) as f:
        content = f.readlines()

    return content


def text(file):
    with open(PATH + file) as f:
        content = [line[:-1] for line in f.readlines()]

    return content


def str_csv(file):
    with open(PATH + file) as csvfile:
        content = [[str(x) for x in rec] for rec in csv.reader(csvfile, delimiter=',')]

    return content


def int_csv(file):
    with open(PATH + file) as csvfile:
        content = [[int(x) for x in rec] for rec in csv.reader(csvfile, delimiter=',')]

    return content


def ints(file):
    with open(PATH + file) as file:
        content = [int(x) for x in file.read().splitlines()]

    data = [int(x) for x in content]
    return data

File: js/common/DataAnalyzer.js
const RESOURCE_ROOT = "../../resources/";

class DataAnalyzer {
    static ints(path) {
        var text = require('fs').readFileSync(RESOURCE_ROOT + path, "utf8").split('\n');
        return text.map( i => parseInt(i, 10) );
    }

    static strs(path) {
        return require('fs').readFileSync(RESOURCE_ROOT + path, "utf8").split('\n');
    }

    static intcsv(path) {
        let data = require('fs').readFileSync(RESOURCE_ROOT + path, "utf8").split('\n');
        let ints = []
        data[0].split(',').forEach(i => {ints.push(parseInt(i))})
        return ints
    }
}

exports.DataAnalyzer = DataAnalyzer
